src/functions/module_math.ts

```diff
@@ -1317,3 +1318,64 @@ export const TRUNC: AddFunctionDescription = {
   },
   isExported: true,
 };
+
+export const MUNIT: AddFunctionDescription = {
+  description: _lt("Return an n*n identity matrix."),
+  args: [arg("n (number)", _lt("size of the unit matrix"))],
+  returns: ["RANGE<NUMBER>"],
+  compute: function (n: PrimitiveArgValue): number[][] {
+    return Array.from({ length: toNumber(n) }, (_, i) =>
+      Array.from({ length: toNumber(n) }, (_, j) => (i === j ? 1 : 0))
+    );
+  },
+  isExported: true,
+};
+
+export const MFILL: AddFunctionDescription = {
+  description: _lt("Return an n*n matrix filled with n."),
+  args: [
+    arg("n (number)", _lt("number of column of the matrix")),
+    arg("m (number)", _lt("number of row of the matrix")),
+    arg("v (number)", _lt("value to fill matrix")),
+  ],
+  returns: ["RANGE<NUMBER>"],
+  compute: function (n: PrimitiveArgValue, m: PrimitiveArgValue, v: PrimitiveArgValue): any[][] {
+    return Array.from({ length: toNumber(n) }, (_, i) =>
+      Array.from({ length: toNumber(m) }, (_, j) => v)
+    );
+  },
+  isExported: true,
+};
+
+export const RANDARRAY: AddFunctionDescription = {
+  description: _lt("Return an n*n identity matrix."),
+  args: [
+    arg("n (number, default=1)", _lt("size of the unit matrix")),
+    arg("m (number, default=1)", _lt("size of the unit matrix")),
+  ],
+  returns: ["RANGE<NUMBER>"],
+  compute: function (n: PrimitiveArgValue = 1, m: PrimitiveArgValue = 1): number[][] {
+    return Array.from({ length: toNumber(n) }, (_, i) =>
+      Array.from({ length: toNumber(m) }, (_, j) => Math.random())
+    );
+  },
+  isExported: true,
+};
+
+function transpose(values, callback: (value: any) => any) {
+  if (isMatrix(values)) {
```

Comment:
we have another generic implementation of a matrix transposition in `transpose2dArray` could be worth looking into it an unify ?

Reply:
The functions implemented in this PR are just there to test the multi cells, they will be dropped before the merge and done in this Pr https://github.com/odoo/o-spreadsheet/pull/2380

I don't remember if I unified things tho, i'll look into it